### DISCUSSION TITLE
Stripe Nodejs SDK

### DISCUSSION THREAD

**Discussion by casualmatt** [nuxt/nuxt#17695] [Questions] [answered]
Hello to all, 

I'm new in coding, almost all the time use low-code tools. 
I'm finding Nuxt pretty easy and i got some help already from the community or documentation, what i didn't find is how to manage SDK or any other nodemodule that i want to use on my server API (/server/api) because all the nuxt3 modules, don't work there. 

Second is how to manage global variable, like if i want to use stripe SDK on my server API , how can i set the API key in just one place, and that it will not be accessible publicly?

ty in advance.
Matt.

**Comment by Diizzayy** [comment, marked as answer]
> how can i set the API key in just one place, and that it will not be accessible publicly?

@casualmatt What you're looking for is [runtime config](http://v3.nuxtjs.org/guide/features/runtime-config)

`nuxt.config.ts`
```ts
export default defineNuxtConfig({
  runtimeConfig: {
    STRIPE_API_KEY: '<STRIPE_API_KEY>'
  }
})
```

> what i didn't find is how to manage SDK or any other nodemodule that i want to use on my server API

As seen by the [Stripe SDK](https://github.com/stripe/stripe-node#usage-with-typescript) example below, almost any sdk can be used. Just be aware of the environment ( browser / server )

`/server/api/stripe.ts`
```ts
import Stripe from 'stripe'

const STRIPE_API_KEY = useRuntimeConfig()?.STRIPE_API_KEY

const stripe = new Stripe(STRIPE_API_KEY, { apiVersion: '2020-08-27' })

export default defineEventHandler((_event) => {
  // handle your logic

  return 'test'
})
```
